bootstrap/api/v1beta1/conversion.go

```diff
@@ -16,79 +16,135 @@ package v1beta1
 import (
 	"fmt"
 
-	"sigs.k8s.io/controller-runtime/pkg/conversion"
+	"k8s.io/apimachinery/pkg/conversion"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
+	ctrlconversion "sigs.k8s.io/controller-runtime/pkg/conversion"
 
-	cabp3v1 "github.com/k3s-io/cluster-api-k3s/bootstrap/api/v1beta2"
+	bootstrapv1beta2 "github.com/k3s-io/cluster-api-k3s/bootstrap/api/v1beta2"
 )
 
 // ConvertTo converts the v1beta1 KThreesConfig receiver to a v1beta2 KThreesConfig.
-func (c *KThreesConfig) ConvertTo(dstRaw conversion.Hub) error {
-	dst := dstRaw.(*cabp3v1.KThreesConfig)
-	if err := autoConvert_v1beta1_KThreesConfig_To_v1beta2_KThreesConfig(c, dst, nil); err != nil {
+func (c *KThreesConfig) ConvertTo(dstRaw ctrlconversion.Hub) error {
+	dst := dstRaw.(*bootstrapv1beta2.KThreesConfig)
+	if err := Convert_v1beta1_KThreesConfig_To_v1beta2_KThreesConfig(c, dst, nil); err != nil {
 		return fmt.Errorf("converting KThreesConfig v1beta1 to v1beta2: %w", err)
 	}
+
+	restored := &bootstrapv1beta2.KThreesConfig{}
+	if ok, err := utilconversion.UnmarshalData(c, restored); err != nil {
```

Comment:
UnmarshalData is just make json deserialization, it doesn't seems like a good way to convert cr version. Shall we get value from variable `c` directly?

Reply:
c may not have the data. 
I believe this is the hub-spoke-hub conversion scenario. So the initial data is stored as annotation in this object, and eventually can be retrieved to restore it.

For example the newly added `CloudProviderName` field would be lost during the conversion without it.

Reply:
cool, thanks